### ISSUE TITLE
function tensorflow::functor::ShuffleInTensor3Simple takes a long time

### ISSUE THREAD

**Issue by XuanBaby** [ROCmSoftwarePlatform/tensorflow-upstream#1186]
<em>Please make sure that this is an issue related to performance of TensorFlow.
As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:performance_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):centos
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):source
- TensorFlow version (use command below):1.15
- Python version:3.6.8
- Bazel version (if compiling from source):0.24.1
- GCC/Compiler version (if compiling from source):7.3.1
- CUDA/cuDNN version: rocm 3.9
- GPU model and memory: 

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I run the ssd_resnet50 network  which fork from :https://github.com/NVIDIA/DeepLearningExamples/tree/master/TensorFlow/Detection/SSD
and use rocprof  hsa-trace found  ShuffleInTensor3Simple takes long time and take up a large proportion
```
"Name","Calls","TotalDurationNs","AverageNs","Percentage"
"void tensorflow::functor::ShuffleInTensor3Simple<float, 2, 1, 0, false>(int, float const*, tensorflow::functor::Dimension<3>, float*) [clone .kd]",3290,309052475,93936,33.4859157800477
8"miopenSp3AsmConv3x3F.kd",1410,153059869,108553,16.58407648293108
"void Eigen::internal::EigenMetaKernel<Eigen::TensorEvaluator<Eigen::TensorAssignOp<Eigen::TensorMap<Eigen::Tensor<int, 1, 1, long>, 16, Eigen::MakePointer>, Eigen::TensorConversionOp<i
nt, Eigen::TensorTupleReducerOp<Eigen::internal::ArgMaxTupleReducer<Eigen::Tuple<long, float> >, Eigen::array<long, 1ul> const, Eigen::TensorMap<Eigen::Tensor<float const, 2, 1, long>, 16, Eigen::MakePointer> const> const> const> const, Eigen::GpuDevice>, long>(Eigen::TensorEvaluator<Eigen::TensorAssignOp<Eigen::TensorMap<Eigen::Tensor<int, 1, 1, long>, 16, Eigen::MakePointer>, Eigen::TensorConversionOp<int, Eigen::TensorTupleReducerOp<Eigen::internal::ArgMaxTupleReducer<Eigen::Tuple<long, float> >, Eigen::array<long, 1ul> const, Eigen::TensorMap<Eigen::Tensor<float const, 2, 1, long>, 16, Eigen::MakePointer> const> const> const> const, Eigen::GpuDevice>, long) [clone .kd]",30,42984065,1432802,4.657334585248345"miopenSp3AsmConvRxSf3x2.kd",209,34859741,166793,3.7770619738291327
```

And  I read the source code ,find maybe this questions abouts threads blocks and __ldg() ... but there are no improvement,
Could you give some advice about how to improved computational performance？Thank you 

**Describe the expected behavior**

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by jerryyin** [comment]
@deven-amd Which repository did we evaluate the resnet50 performance? Was this `ShuffleInTensor3Simple` kernel's performance an issue that needs to address? I doubt this is relevant but wants to double check.

**Comment by deven-amd** [comment]
we used TF from `develop-upstream` in our repo + latest ROCm (3.9) for our perf evals. Conv and BatchNorm were the top-two buckets

As per the data posted in the description, the ` ShuffleInTensor3Simple` routine is taking 33% of the time! that is not what we saw in our measurements. I would recommend trying a newer TF version (description mentions 1.15, which is quite old at this point).

**Comment by jerryyin** [comment]
Closing the issue for inactivity. Feel free to comment if issue persists.
